sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala

```diff
@@ -111,30 +113,37 @@ class SerializeConcatHostBuffersDeserializeBatch(
       }
       assert(headers.length <= 1 && buffers.length <= 1)
       withResource(new NvtxRange("broadcast manifest batch", NvtxColor.PURPLE)) { _ =>
-        if (headers.isEmpty) {
-          batchInternal = GpuColumnVector.emptyBatchFromTypes(dataTypes)
-          GpuColumnVector.extractBases(batchInternal).foreach(_.noWarnLeakExpected())
-        } else {
-          withResource(JCudfSerialization.readTableFrom(headers.head, buffers.head)) { tableInfo =>
-            val table = tableInfo.getContiguousTable
-            if (table == null) {
-              val numRows = tableInfo.getNumRows
-              batchInternal = new ColumnarBatch(new Array[ColumnVector](0), numRows)
-            } else {
-              batchInternal = GpuColumnVectorFromBuffer.from(table, dataTypes)
-              GpuColumnVector.extractBases(batchInternal).foreach(_.noWarnLeakExpected())
-              table.getBuffer.noWarnLeakExpected()
+        try {
+          val res = if (headers.isEmpty) {
+            val gpuCV = GpuColumnVector.emptyBatchFromTypes(dataTypes)
+            GpuColumnVector.extractBases(gpuCV).foreach(_.noWarnLeakExpected())
+            Left(gpuCV)
+          } else {
+            withResource(JCudfSerialization.readTableFrom(headers.head, buffers.head)) {
+              tableInfo =>
+                val table = tableInfo.getContiguousTable
+                if (table == null) {
+                  val numRows = tableInfo.getNumRows
+                  Left(new ColumnarBatch(new Array[ColumnVector](0), numRows))
+                } else {
+                  withResource(tableInfo) { _ =>
+                    val spillableBatch = SpillableColumnarBatch(table, dataTypes,
+                      SpillPriorities.ACTIVE_BATCHING_PRIORITY, RapidsBuffer.defaultSpillCallback)
+                    Right(spillableBatch)
+                  }
+                }
             }
           }
+          batchInternal = res
+          res
+        } finally {
+          headers = null
+          buffers = null
+          // At this point we no longer need the host data and should not need to touch it again.
+          buffers.safeClose() // keep last to avoid a nested finally
```

Comment:
This is a useless operation if we just set buffers to null

Reply:
incorrect change indeed, I'll put it back in the original order